### ISSUE TITLE
False positive needless_lifetimes

### ISSUE THREAD

**Issue by alexander-irbis** [intellij-rust/intellij-rust#3541]
* Rust plugin version: v0.2.93.2117-191

## Problem description

The linter doesn't take into account usages of the lifetime in the fn body.

```rust
fn f<'i>(input: In<'i>) -> Out<'i> {
    // ... a closure in the fn body that requires an explicit type with lifetime:
        |mut acc: Out<'i>, item| {
            acc.0.push(item);
            acc
        }
    // ...
}
```

**Comment by vlad20012** [comment]
I can't reproduce
![Peek 2019-03-15 17-38](https://user-images.githubusercontent.com/3221931/54439084-262a4580-4749-11e9-8dec-c3e671312936.gif)

**Comment by alexander-irbis** [comment]
Ok, sorry, I oversimplified the example.
There is involved a nom macros in my code.
So, maybe this issue should be closed.
